File: ui/base/src/main/java/org/cru/godtools/base/ui/util/DrawableUtils.java
package org.cru.godtools.base.ui.util;

import android.graphics.drawable.Drawable;

import androidx.annotation.ColorInt;
import androidx.annotation.Nullable;
import androidx.core.graphics.drawable.DrawableCompat;

public class DrawableUtils {
    @Nullable
    public static Drawable tint(@Nullable final Drawable drawable, @ColorInt final int color) {
        if (drawable != null) {
            final Drawable out = DrawableCompat.wrap(drawable).mutate();
            DrawableCompat.setTint(out, color);
            return out;
        }
        return null;
    }
}

File: ui/base/src/main/kotlin/org/cru/godtools/base/ui/util/DrawableUtils.kt
package org.cru.godtools.base.ui.util

import android.graphics.drawable.Drawable
import androidx.annotation.ColorInt
import androidx.core.graphics.drawable.DrawableCompat

fun Drawable?.tint(@ColorInt color: Int) =
    this?.let { DrawableCompat.wrap(it).mutate() }?.apply { DrawableCompat.setTint(this, color) }
